src/utils/walletMigrationArchanova.js

```diff
@@ -0,0 +1,299 @@
+      balance: maxEthToMigrate.toFixed(6, BigNumber.ROUND_DOWN),
+      decimals: eth.decimals,
+    },
+  };
+}
+
+const buildAssetMigrationRawTransactions = async (
+  wallet: Wallet,
+  accounts: Account[],
+  tokensToMigrate: TokensToMigrateByAddress,
+  collectiblesToMigrate: CollectibleId[],
+): Promise<string[]> => {
+  const etherspotAccount = findFirstEtherspotAccount(accounts);
+  if (!etherspotAccount) {
+    reportErrorLog('buildAssetMigrationRawTransactions: no etherspot account found');
+    throw new Error(t('error.failedToBuildTransaction'));
+  }
+
+  const archanovaAccount = findFirstArchanovaAccount(accounts);
+  if (!archanovaAccount) {
+    reportErrorLog('buildAssetMigrationRawTransactions: no etherspot account found');
+    throw new Error(t('error.failedToBuildTransaction'));
+  }
+
+  const migratorParams = {
+    chainId: mapChainToChainId(CHAIN.ETHEREUM),
+    archanovaAccount: getAccountAddress(archanovaAccount),
+    etherspotAccount: getAccountAddress(etherspotAccount),
+  };
+  logBreadcrumb('walletMigrationArchanova', 'building migrator', migratorParams);
+  const migrator = new Migrator(migratorParams);
+
+  logBreadcrumb('walletMigrationArchanova', 'apply add migrator device if needed');
+  await applyAddMigratorDeviceTransactionIfNeeded(migrator);
+
+  logBreadcrumb('walletMigrationArchanova', 'apply transfer transactions');
+  await applyAssetTransferTransaction(migrator, tokensToMigrate, collectiblesToMigrate);
+
+  // Sign migration message
+  logBreadcrumb('walletMigrationArchanova', 'sign message');
+  const archanovaAccountDeviceSignature = await wallet.signMessage(migrator.migrationMessage);
+  if (!archanovaAccountDeviceSignature) {
+    reportErrorLog('buildAssetMigrationRawTransactions: signMessage returned null');
+    throw new Error(t('error.failedToSignMessage'));
+  }
+
+  logBreadcrumb('walletMigrationArchanova', 'encode transaction request');
+  return migrator.encodeTransactionRequests(archanovaAccountDeviceSignature).map(({ data, to }) => {
+    if (addressesEqual(to, migrator.migratorAddress)) {
+      try {
+        const { account } = archanovaService.getSdk().contract;
+        return account.encodeMethodInput('executeTransaction', to, 0, data);
+      } catch (error) {
+        reportErrorLog('buildAssetMigrationRawTransactions: encodeMethodInput failed', {
+          error,
+          encodedTransactionRequest: { to, data },
+          migratorAddress: migrator.migratorAddress,
+          archanovaAccount: archanovaAccount.id,
+          etherspotAccount: etherspotAccount.id,
+        });
+        throw new Error(t('error.failedToBuildTransaction'));
+      }
+    }
+
+    return data;
+  });
+};
+
+async function applyAddMigratorDeviceTransactionIfNeeded(migrator: Migrator): Migrator {
+  const migratorDevice = await archanovaService.getConnectedAccountDevice(migrator.migratorAddress);
+  const isMigratorDeviceDeployed = migratorDevice?.state === 'Deployed';
```

Comment:
Can use `isArchanovaDeviceDeployed`

Reply:
`isArchanovaDeviceDeployed` not only checks `state === Deployed` but also returns true when `nextState === Deployed`. Not sure what is the meaning of `nextState` in`ArchanovaAccountDevice`. wdyt?

Reply:
`nextState` means that account device is being deployed.